homeassistant/components/streamlabswater/binary_sensor.py

```diff
@@ -0,0 +1,73 @@
+"""Support for Streamlabs Water Monitor Away Mode."""
```

Comment:
Why not implement this as a switch and get rid of the `streamlabswater.set_away_mode`service?

Reply:
Two reasons: 1.) When I first started to implement this I considered that but then noticed that would cause it to be controlled along with all other switches in a call to `turn_off` switches when not specifying a particular entity_id (via stuff a user might have set up programatically or via the one click button in the UI to toggle all switches). This seemed pretty risky to me in that disabling away_mode unintentionally could potentially mean a user's home is not being monitored for water user while gone and a leak would go undetected.  So, I thought it was worth making it a separate service so a user could trust it would be operated on independently and integrated in a way.

2.) In general when I surveyed the various switches in home assistant nearly all of them represented turning on or off some physical entity in the real world rather than a conceptual behavior of the entity...so while technically it applicable (it's just a togglable binary value after all) it seemed like it did not fit with the intent of what a switch is.

Reply:
You are absolutely right! But if you build it as an alarm_panel it would make sense, arm away and arm home matches the service call & you can show the triggered state instead of the binary sensor.

Reply:
Right, that makes sense. Long term I think it would be cool for the away mode to integrate with some presence system supported in the core of home assistant so that it gets managed automatically for the user. For now someone could use the [switch template](https://www.home-assistant.io/components/switch.template/) if they wanted it to act like a switch.

Reply:
The "integration with presence system" is done via automations like this one:

```yaml
automation:
  trigger:
  - platform: zone
    entity_id: device_tracker.paulus
    zone: zone.home
    event: leave
  action:
  - service: alarm_control_panel.alarm_arm_away
    data:
      entity_id: alarm_control_panel.streamlabs_water
```